go-controller/pkg/ovn/pods.go

```diff
@@ -26,24 +26,38 @@ import (
 )
 
 func (oc *Controller) syncPods(pods []interface{}) {
+	err := wait.PollImmediate(500*time.Millisecond, 60*time.Second, func() (bool, error) {
+		if err := oc.syncPodsRetriable(pods); err != nil {
+			klog.Errorf("Failed (will retry) in syncing pods: %v", err)
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		klog.Fatalf("Error in syncing pods: %v", err)
+	}
+}
+
+func (oc *Controller) syncPodsRetriable(pods []interface{}) error {
```

Comment:
a good opportunity to add docstring ?

Reply:
How about
```
// This function implements the main body of work of syncPods.
// Upon failure, it may be invoked multiple times in order to avoid a pod restart.
```